InteroperabilityTest_Engine/Compute/Compare.cs

```diff
@@ -0,0 +1,388 @@
+                            Message = "Data is different from the reference, worse than reference.",
+                            Status = oM.Test.TestStatus.Error,
+                            Property = result.PropertyID,
+                            RunValue = result.ReturnedItem,
+                            ReferenceValue = reference.ReturnedItem
+                        };
+                    }
+                }
+                else
+                {
+                    //Can not extract numerical information, report back that there is a difference
+                    return new ComparisonDifference
+                    {
+                        Message = "Data is different from the reference",
+                        Status = oM.Test.TestStatus.Error,
+                        Property = result.PropertyID,
+                        RunValue = result.ReturnedItem,
+                        ReferenceValue = reference.ReturnedItem
+                    };
+                }
+            }
+        }
+
+
+        /***************************************************/
+        /**** Public Methods - Interface                ****/
+        /***************************************************/
+
+        [Description("Compares two TestInformation results against each other by dispatching to a comparison matching the type of the two obejcts. Types between the result and reference need to be an exact match.")]
+        [Input("result", "Result from a test just run to be compared to a reference result.")]
+        [Input("refResult", "Reference result to compare to. Needs to be of the same type as the provided result.")]
+        [Output("result", "A test result containing information about the equality of the two input TestInformations. Returned type will depend on provided type.")]
+        public static oM.Test.ITestInformation ICompare(oM.Test.ITestInformation result, oM.Test.ITestInformation refResult)
+        {
+            if (result == null)
+                return null;
+
+            if (refResult == null)
+                return result.INoReferenceFound();
+
+            return Compare(result as dynamic, refResult as dynamic);
+        }
+
+        /***************************************************/
+        /**** Private Methods - Fallback                ****/
+        /***************************************************/
+
+        private static oM.Test.ITestInformation Compare(oM.Test.ITestInformation result, oM.Test.ITestInformation refResult)
+        {
+            return null;
+        }
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private static oM.Test.ITestInformation IGetSameIDInformation(this IEnumerable<oM.Test.ITestInformation> infoList, oM.Test.ITestInformation toFind)
+        {
+            return GetSameIDInformation(infoList as dynamic, toFind as dynamic);
```

Comment:
@IsakNaslundBh do we want to include fall back methods for this in case future objects implement the `ITestInformation` interface and come through here?

Reply:
It actually already has an implicit fall back method. The default implementation on line 365-368 already accepts `ITestInformation` and calls `IHasMatchingIDs` which has a fall back, so should not crash for anything (for that reason at least!)